### ISSUE TITLE
Ollama API error if OpenAI API key is not specified

### ISSUE THREAD

**Issue by arcaneum** [joaomdmoura/crewAI#387]
@joaomdmoura Just wondering whether you had a chance to look at this further?

I'm also running into the same issue whereby the code throws an error if the OpenAI API key is not specified, in spite of pointing to Ollama server. And if I provide the OpenAI API key at an environment level and run the code it works, but it is consuming tokens from my OpenAI API account rather than using the Ollama endpoint, even though no part of my code points to OpenAI, other than the environment level API Key. 

Basically API's for Ollama, LMStudio, Groq etc. (I have tested on all of those) are not functional without the OpenAI API key and I'm not able to figure out from where the token usage is 'leaking' into my OpenAI account since my code has nothing to that effect.

Please can you help?

_Originally posted by @arcaneum in https://github.com/joaomdmoura/crewAI/issues/324#issuecomment-2022576364_

**Comment by arcaneum** [comment]
Adding more context. 

### This is the part of code that points to Ollama API:

![image](https://github.com/joaomdmoura/crewAI/assets/38038031/f6428ee8-6be0-429e-934c-14f63b80e572)


### And the OpenAI API not found error I'm getting:

Traceback (most recent call last):
  File "/Users/<<>>/AI/Coding/CrewAI/CrewAI_Storybook_Automation_My_Version/Story_Book_CrewAI.py", line 32, in <module>
    llm = ChatOpenAI(
          ^^^^^^^^^^^
  File "/Users/<<>>/miniconda3/lib/python3.11/site-packages/langchain_core/load/serializable.py", line 120, in __init__
    super().__init__(**kwargs)
  File "/Users/<<>>/miniconda3/lib/python3.11/site-packages/pydantic/v1/main.py", line 341, in __init__
    raise validation_error
pydantic.v1.error_wrappers.ValidationError: 1 validation error for ChatOpenAI
__root__
  Did not find openai_api_key, please add an environment variable `OPENAI_API_KEY` which contains it, or pass `openai_api_key` as a named parameter. (type=value_error)

**Comment by IvanLee113** [comment]
make sure your script load env file 

```
from dotenv import load_dotenv
load_dotenv()
```

**Comment by arcaneum** [comment]
> ```
> llm = ChatOpenAI(openai_api_base=os.environ.get("OPENAI_API_BASE"),
>                         openai_api_key=os.environ.get("OPENAI_API_KEY"),                    
>                         model_name=os.environ.get("OPENAI_MODEL_NAME"))
> ```
> 
> in .env file
> 
> ```
> OPENAI_API_KEY=NA
> OPENAI_API_BASE='http://localhost:11434/v1'
> OPENAI_MODEL_NAME='llama2'
> ```
> 
> Above works fine for me, try it out

@IvanLee113, Excellent! That worked for me! Thank you very much for your help! Much appreciated! 😊
